test-ets.sh

```diff
@@ -1,32 +1,52 @@
-# this script is meant to run the Ethereum Test Suite (https://github.com/ethereum/tests) on CircleCI
-# to skip these tests, set SKIP_ETS_TEST variable CircleCI UI
-# to differentiate by branch, use CIRCLE_BRANCH variable in the conditional below
-
-echo "current branch: $BUILDKITE_BRANCH"
-
-echo "memory:"
-free -m
+#!/usr/bin/env bash
 
 if [ -z "$IN_NIX_SHELL" ]; then
     export SBT_NIX="-Dnix=true"
 fi
 
-# at least 1.5G of heap space and 10M of stack size is required for ETS.
-export JAVA_OPTS="-Xmx2g -Xss16m -XX:MaxMetaspaceSize=512m"
-export SBT = "sbt -v -mem 2048 $SBT_NIX";
-
-if [ -z $SKIP_ETS_TESTS ]; then
-    git submodule init;
-    git submodule update;
-    if [ "$BUILDKITE_BRANCH" == "master" -o -n "$RUN_FULL_ETS" ]; then
-        export JAVA_OPTS="-Xmx3g -Xss16m -XX:MaxMetaspaceSize=512m"
-        echo "running full ETS"
-        $SBT "ets:testOnly * -- -DuseLocalVM=true -Dexg=vmPerf*";
-    else
-        echo "running a subset of ETS"
-        $SBT "ets:testOnly *VMSuite -- -Dexg=vmPerf*" &&
-        $SBT "ets:testOnly *BlockchainSuite -- -DuseLocalVM=true -Ding=bcForkStress*,bcMulti*,bcState*,bcTotalDiff*,bcValidBlock*,Transition*";
-    fi
-else
-    echo "SKIP_ETS_TESTS variable is set - skipping the tests";
-fi
+git submodule init
+git submodule update
+
+echo "booting Mantis and waiting for RPC API to be up"
+$SBT -Dconfig.file=./src/main/resources/conf/testmode.conf run &> mantis-log.txt &
+
+while ! nc -z localhost 8546; do   
+  sleep 0.1
```

Comment:
suggestion: we have a GET '/healthcheck' endpoint

Reply:
not for this PR but might prove more reliable

Reply:
Not sure if that's a good fit. Giving it a try right now locally in testmode I get a status 500 response:

```json
{
  "checks":[
    {
      "description":"listening",
      "status":"OK"
    },
    {
      "description":"peerCount",
      "status":"OK"
    },
    {
      "description":"earliestBlock",
      "status":"OK"
    },
    {
      "description":"latestBlock",
      "status":"OK"
    },
    {
      "description":"pendingBlock",
      "status":"ERROR",
      "error":"java.util.NoSuchElementException: None.get"
    }
  ]
}
```

Reply:
To clarify: when something breaks this health check endpoint the result would be a stuck CI build.

Reply:
isn't the same true for this while loop if the port does not become open?

Reply:
Yes, but the actual functionality (eg. the pending block) might well be broken, which the healthcheck will test. Only the HTTP server needs to be up.

Reply:
clear.